synapse/handlers/federation.py

```diff
@@ -1460,9 +1493,21 @@ async def _sync_partial_state_room(
         # TODO(faster_joins): what happens if we leave the room during a resync? if we
         #   really leave, that might mean we have difficulty getting the room state over
         #   federation.
-        #
-        # TODO(faster_joins): try other destinations if the one we have fails
 
+        # Make an infinite iterator of destinations to try. Once we find a working
+        # destination, we'll stick with it until it flakes.
```

Comment:
On a restart, it might take a little time before we stumble upon a working remote server.